options/posix/generic/posix_string.cpp

```diff
@@ -132,9 +132,8 @@ char *strcasestr(const char *s, const char *pattern) {
 	return nullptr;
 }
 
-char *strdupa(const char *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+char *strdupa(const char *str) {
+	return strcpy(reinterpret_cast<char *>(alloca(strlen(str) + 1)), str);
```

Comment:
is this code actually safe? from `man alloca`:
> This temporary space is automatically freed when the function that called alloca() returns to its caller.

In this case I would expect the destination buffer to be freed upon `strdupa` returning - for the sake of completeness you should write a test case specificially engineered to use the stack anyway, such that improper use of `alloca` will be detected by seeing that the destination buffer was corrupted

Reply:
no